engine: create fix for long code wrap (#1307)
redpanda-data/documentation: modules/deploy/pages/deployment-option/self-hosted/kubernetes/k-25.2-beta.adoc

= Try the 25.2 Beta of the Redpanda Operator
:page-beta: true

> :description: Deploy the 25.2 beta release of the Redpanda Operator. This version of the Redpanda Operator defaults to cluster scope so a single instance of the Operator can manage multiple Redpanda resources in different namespaces.

Reviewer: "defaults" here feels misleading relative to [this PR](https://github.com/redpanda-data/docs/pull/1308) where you say support for multiple operators is removed. 

Maybe just "is cluster scoped"?
```suggestion
:description: Deploy the 25.2 beta release of the Redpanda Operator. This version of the Redpanda Operator is cluster scope so a single instance of the Operator can manage multiple Redpanda resources in different namespaces.
```

```
+
[.no-copy]
----
deployment "redpanda-controller-operator" successfully rolled out
----

. Install a xref:reference:k-crd.adoc[Redpanda custom resource] to deploy a Redpanda cluster.
+
.`redpanda-cluster.yaml`
[,yaml,subs="attributes+"]
----
apiVersion: cluster.redpanda.com/v1alpha2
kind: Redpanda
metadata:
  name: redpanda
----

. If you want to use enterprise features in Redpanda, add the details of a Secret that stores your Enterprise Edition license key.
+
.`redpanda-cluster.yaml`
[,yaml,subs="attributes+"]
----
apiVersion: cluster.redpanda.com/v1alpha2
kind: Redpanda
metadata:
  name: redpanda
spec:
  clusterSpec:
    enterprise:
      licenseSecretRef:
        name: <secret-name>
        key: <secret-key>
----
+
For details, see xref:get-started:licensing/add-license-redpanda/kubernetes.adoc[].

. Apply the Redpanda resource:
+
```bash
kubectl apply -f redpanda-cluster.yaml --namespace <namespace>
```

. Wait for the Redpanda Operator to deploy the cluster:
+
```bash
kubectl get redpanda --namespace <namespace> --watch
```
+
[.no-copy]
----
NAME       READY   STATUS
redpanda   True    Redpanda reconciliation succeeded
----
+

> This step may take a few minutes. You can watch for new Pods to make sure that the deployment is progressing:

Reviewer: It should be pretty snappy. Is that not what you've experienced?

Reviewer 2: It sometimes takes a while for the cluster to be ready. I've waited multiple minutes before.

Reviewer: Ah, for the cluster to be ready. Do you often see the redpanda Pods handing around in a Scheduled but not yet Running state?

I've seen two primary sources of latency:
- cert-manager can be a bit sluggish which causes Kubelet's backoff loops to lengthen because it's waiting for the certs to be minted
- kubelet sometimes takes >1m to actually start running Pods :/ I haven't been able to figure out why but it's quite frustrating. There's even a dedicated PodStartLatency SLO metric that kubelet emits where it more or less "tells on itself".
